### PR TITLE
Make iterations metric per rule_group

### PR DESCRIPTION
**Pull request by gouthamve**
So I have a feeling we didn't do `rule_group` for those metrics on purpose. The metric output doesn't have the iterations metrics anymore.

### CODE REVIEW

**Comment by gouthamve** [comment]
The reason we didn't do that before is because when we first made this change, upstream Prometheus metric didn't have a `rule_group` metric.

**Comment by gouthamve** [comment]
Can you please rebase against master to pull in https://github.com/cortexproject/cortex/pull/4137 and move the changelog entry to the top?

File: pkg/util/metrics_helper.go [hunk 1]
```diff
@@ -113,6 +113,7 @@ func (mfm MetricFamilyMap) SumSummariesTo(name string, output *SummaryData) {
 
 func (mfm MetricFamilyMap) sumOfSingleValuesWithLabels(metric string, labelNames []string, extractFn func(*dto.Metric) float64, aggregateFn func(labelsKey string, labelValues []string, value float64)) {
 	metricsPerLabelValue := getMetricsWithLabelNames(mfm[metric], labelNames)
+	fmt.Println(metricsPerLabelValue, metric, labelNames)
```
Comment (pstibrany) [inline pkg/util/metrics_helper.go:116]: Can we remove this?

**Review by gotjosh — APPROVED** [review]
LGTM

**Review by pracucci — COMMENTED** [review]
@gouthamve Could you open a PR with the CHANGELOG entry for this PR change, please?

**Pull request merged**